packages/bitcoin-swap-provider/lib/BitcoinSwapProvider.ts

```diff
@@ -37,7 +37,7 @@ export default class BitcoinSwapProvider extends Provider implements Partial<Swa
       throw new Error(`Mode must be one of ${swapModes.join(',')}`)
     }
     this._network = network
-    this._mode = mode
+    this._mode = network.segwitCapable ? mode : bitcoin.SwapMode.P2SH
```

Comment:
Similar to the other point about throwing an error if the wrong type was supplied